### ISSUE TITLE
False negative used-before-assignment for self-referencing assignment in if-statement

### ISSUE THREAD

**Issue by snordhausen** [PyCQA/pylint#6643]
### Bug description

The following code  should trigger a `used-before-assignment` error from pylint, but it does not:

```
if 1 + 1 == 2:
    x = x + 1
```

When the if-statement is removed like
```
x = x + 1
```
the error is produced as expected.

### Command used

```shell
python -m pylint foobar.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

There should be a `used-before-assignment` error.

### Pylint version

```shell
$ python -m pylint --version
pylint 2.13.9
astroid 2.11.5
Python 3.8.13 (default, Mar 31 2022, 14:08:37)
```

### OS / Environment

Python interpreter was built by pyenv, on Ubuntu Linux 20.04

**Comment by mbyrnepr2** [comment]
Duplicate of https://github.com/PyCQA/pylint/issues/4913 & https://github.com/PyCQA/pylint/issues/3229?

**Comment by jacobtylerwalls** [comment]
This one is also a leeetle different from #1727, because the problem usage is _inside_ the conditional, not after it, so it doesn't depend on control flow--it's just totally wrong.
